lnrpc/wtclientrpc/wtclient.go

```diff
@@ -336,33 +336,33 @@ func (c *WatchtowerClient) GetTowerInfo(ctx context.Context,
 // returns a map of acked-update counts and one for un-acked-update counts that
 // will be populated once the db call has been made.
 func constructFunctionalOptions(includeSessions bool) (
-	[]wtdb.ClientSessionListOption, map[wtdb.SessionID]uint16,
+	[]wtdb.ClientSessionListOption, map[wtdb.SessionID]uint64,
 	map[wtdb.SessionID]uint16) {
 
 	var (
 		opts                  []wtdb.ClientSessionListOption
-		ackCounts             = make(map[wtdb.SessionID]uint16)
 		committedUpdateCounts = make(map[wtdb.SessionID]uint16)
+		ackCounts             = make(map[wtdb.SessionID]uint64)
 	)
 	if !includeSessions {
 		return opts, ackCounts, committedUpdateCounts
 	}
 
-	perAckedUpdate := func(s *wtdb.ClientSession, _ uint16,
-		_ wtdb.BackupID) {
+	perNumAckedUpdates := func(s *wtdb.ClientSession, id lnwire.ChannelID,
+		numUpdates uint64) {
 
-		ackCounts[s.ID]++
+		ackCounts[s.ID] += numUpdates
 	}
 
 	perCommittedUpdate := func(s *wtdb.ClientSession,
-		_ *wtdb.CommittedUpdate) {
+		u *wtdb.CommittedUpdate) {
 
 		committedUpdateCounts[s.ID]++
 	}
 
 	opts = []wtdb.ClientSessionListOption{
-		wtdb.WithPerAckedUpdate(perAckedUpdate),
 		wtdb.WithPerCommittedUpdate(perCommittedUpdate),
+		wtdb.WithPerNumAckedUpdates(perNumAckedUpdates),
```

Comment:
Is the ordering relevant here?

Reply:
on no the ordering is not. This is actually removing an old option though and adding a new one. But can change to add the new one in the same position as the one being removed :+1: